### PR TITLE
helloworld: note printf->puts optimization

### PR DESCRIPTION
**Pull request by brooksdavis**
Note the optimization may happen and update the example output
to match.

### CODE REVIEW

File: src/helloworld/README.md [hunk 1]
```diff
@@ -133,15 +133,16 @@ Breakpoint 1, main () at helloworld.c:6
 (gdb) 
 ```
 
-If you then step into the `printf` call you will see that GDB prints
-the capability argument with expanded information including bounds and
-permissions:
+If you then step, you into the `printf` (or `puts` depending on
```
Comment (jrtc27) [inline src/helloworld/README.md:136]: Something odd here

File: src/helloworld/README.md [hunk 2]
```diff
@@ -133,15 +133,16 @@ Breakpoint 1, main () at helloworld.c:6
...
+optimization level) call you will see that GDB prints the capability
+argument with expanded information including bounds and permissions:
 
 ```
 (gdb) s
-printf (fmt=0x1006e0 [rR,0x1006e0-0x1006ed] "Hello world\n")
-    at /home/bed22/cheri/cheribsd/lib/libc/stdio/printf.c:57
-57      /home/bed22/cheri/cheribsd/lib/libc/stdio/printf.c: No such file or directory.
+(gdb) s
```
Comment (jrtc27) [inline src/helloworld/README.md:142]: Is the intent to show both or just one? This has two steps

File: src/helloworld/README.md (hunk 1 above)
Comment (brooksdavis) [reply src/helloworld/README.md]: hopefully more sensible now

File: src/helloworld/README.md (hunk 2 above)
Comment (brooksdavis) [reply src/helloworld/README.md]: Not quite enough replacement when pasting...

**Review by jrtc27 — APPROVED** [review]
Maybe a little strange still that we say "printf (or puts)" when puts is expected based on the optimisation level suggested above and shown in the GDB output, but eh

**Review by rwatson — APPROVED** [review]

**Pull request merged**